packages/pluggableWidgets/groupbox-native/src/GroupBox.tsx

```diff
@@ -0,0 +1,140 @@
+import { createElement, ReactElement, ReactNode, useCallback, useEffect, useRef, useState } from "react";
+import { Animated, Easing, LayoutChangeEvent, Pressable, View, ViewStyle } from "react-native";
+import { Icon } from "mendix/components/native/Icon";
+import { exclude, flattenStyles } from "@mendix/piw-native-utils-internal";
+
+import { GroupBoxStyle, defaultGroupBoxStyle } from "./ui/Styles";
+import { GroupBoxProps } from "../typings/GroupBoxProps";
+
+declare interface GlyphIcon {
+    readonly type: "glyph";
+    readonly iconClass: string;
+}
+
+export function GroupBox(props: GroupBoxProps<GroupBoxStyle>): ReactElement | null {
+    const styles = flattenStyles(defaultGroupBoxStyle, props.style);
+    const startedExpanded = !props.showHeader || props.collapsible === "collapsibleYesExpanded";
+    const [isExpanded, setIsExpanded] = useState(startedExpanded);
+
+    const toggleContent = () => {
+        setIsExpanded(value => !value);
+    };
+
+    const renderIcon = () => {
+        const { iconCollapsed, iconExpanded } = props;
+        const customIconsConfigured = iconCollapsed && iconCollapsed.value && iconExpanded && iconExpanded.value;
+        const customIconSource = iconCollapsed?.value;
+        const customExpandedIconSource = iconExpanded?.value;
+        const source = isExpanded ? customExpandedIconSource : customIconSource;
+        const iconStyles = exclude(styles.header.icon, ["size", "color"]);
+
+        return customIconsConfigured ? (
+            <View style={iconStyles}>
```

Comment:
Dont we want to make the custom icons animated aswell ?

Reply:
Don't think we can. I tried using a morph library but that didn't look good.
We don't know what custom icons will be used so we don't know how it should be animated